packages/portalnetwork/src/wire/utp/Packets/Packet.ts

```diff
@@ -25,29 +26,27 @@ export class Packet {
   extensions: any[]
 
   public static bufferToPacket(buffer: Buffer): Packet {
-    const ptandver = buffer[0].toString(16)
-    const ver = ptandver[1]
-    const _version = parseInt(ver, 16)
     const extension = buffer.readUInt8(1)
     let packet: Packet
     if (extension === 1) {
       const size = buffer.readUInt8(21)
+      const bitmask = buffer.subarray(22, 22 + size)
       packet = new Packet({
         header: new SelectiveAckHeader(
           {
             pType: buffer[0] >> 4,
             version: 1,
             extension: buffer.readUInt8(1),
             connectionId: buffer.readUInt16BE(2),
-            timestampMicroseconds: buffer.readUInt32BE(4),
-            timestampDifferenceMicroseconds: buffer.readUInt32BE(8),
-            wndSize: buffer.readUInt32BE(12),
+            timestampMicroseconds: BigNumber.from(buffer.subarray(4, 8)),
+            timestampDifferenceMicroseconds: BigNumber.from(buffer.subarray(8, 12)),
+            wndSize: BigNumber.from(buffer.readUInt32BE(12)),
             seqNr: buffer.readUInt16BE(16),
             ackNr: buffer.readUInt16BE(18),
           },
-          buffer.subarray(22, 22 + size)
+          bitmask
         ),
-        payload: buffer.subarray(22 + size),
+        payload: Uint8Array.from([]),
```

Comment:
Just to confirm my understanding, there's no payload with a selective Ack?

Reply:
correct, there is a header-extension with a bitmask, but no payload